### PR TITLE
limit memory in sts debug mode

### PR DESCRIPTION
**Pull request by aryan16**
**Please provide a description of this PR:**

### CODE REVIEW

File: security/pkg/stsservice/server/server.go [hunk 1]
```diff
@@ -135,7 +136,10 @@ func (s *Server) validateStsRequest(req *http.Request) (security.StsRequestParam
 	}
 
 	if stsServerLog.DebugEnabled() {
-		reqDump, _ := httputil.DumpRequest(req, true)
+		debugReq := req
+		// Limit req body to 10 mb
+		debugReq.Body = io.NopCloser(io.LimitReader(req.Body, 1024*1024*10))
```
Comment (howardjohn) [inline security/pkg/stsservice/server/server.go:141]: the NopCloser seems problematic? I think we need to actually close it?

http.MaxBytesReader may be more appropriate

Comment (aryan16) [reply security/pkg/stsservice/server/server.go:141]: sgtm, just checked `NopClosed` is depreceated in golang.

**Review by lei-tang — COMMENTED** [review]
LGTM. Will let John to take another look before approving.

**Review by howardjohn — APPROVED** [review]

**Review by lei-tang — APPROVED** [review]

**Pull request merged**